python/social_bot/envs/grocery_ground.py

```diff
@@ -191,20 +207,49 @@ def run(self, agent, world):
             agent (pygazebo.Agent): the learning agent 
             world (pygazebo.World): the simulation world
         """
-        pre_agent_pos = self.task_specific_observation()[:2]
+        self._pre_agent_pos = self.get_icub_extra_obs(agent)[:3]
         agent_sentence = yield
         done = False
+        # set icub initial pose
+        x = self._agent_init_pos[0] + random.random() * self._random_range
+        y = self._agent_init_pos[1] + random.random() * self._random_range
+        agent_pos = np.array([x, y, 0.6])
+        # a trick from roboschool, to encourage optimal straight walk
+        orient = 0
+        if random.randint(0, 2) == 0:
```

Comment:
This trick comes form [roboschool_humanoid_flagrun](https://github.com/openai/roboschool/blob/42d1c1c14f57b58f156072b6b5f9b761ecaf3018/roboschool/gym_humanoid_flagrun.py#L22)
It is helpful to learn to steer and walk straightly, and speed up the learning progress a lot. The agent can walk to the ball only after about 2 hours training.